app/GHCi/DAP/Command.hs

```diff
@@ -1122,7 +1123,7 @@ sourceCmd_ :: D.SourceRequestArguments
           -> Gi.GHCi (Either String D.SourceResponseBody)
 sourceCmd_ args = do
   modSums <- Gi.getLoadedModules
-  let Just srcInfo =  D.sourceSourceRequestArguments args
+  let srcInfo = fromJust $ D.sourceSourceRequestArguments args
```

Comment:
Ignore the case which D.sourceSourceRequest Arguments returns Nothing because the latest implementation is expected to return source instead of sourceReference in SourceArguments.
https://microsoft.github.io/debug-adapter-protocol/specification#Requests_Source
```
interface SourceArguments {
  /**
   * Specifies the source content to load. Either `source.path` or
   * `source.sourceReference` must be specified.
   */
  source?: Source;

  /**
   * The reference to the source. This is the same as `source.sourceReference`.
   * This is provided for backward compatibility since old clients do not
   * understand the `source` attribute.
   */
  sourceReference: number;
}
```

Should I use pattern matching and throw an exception when D.sourceSourceRequestArguments return Nothing?

Reply:
Thanks for the PR.
I tested and it's caught like bellow.
It would be nice to throw an exception with a message such a "deprecated data,,, not supported..."

let Just srcInfo =  D.sourceSourceRequestArguments args
```
H>>= :dap-source ...
H>>= 
app/GHCi/DAP/Command.hs:1127:7-99: Non-exhaustive patterns in Just srcInfo

:dap-scopes ...
H>>= :dap-variables ...
```

let srcInfo = fromJust $ D.sourceSourceRequestArguments args
```
H>>= :dap-source ...
H>>= 
Maybe.fromJust: Nothing
CallStack (from HasCallStack):
  error, called at libraries/base/Data/Maybe.hs:148:21 in base:Data.Maybe
  fromJust, called at app/GHCi/DAP/Command.hs:1128:17 in main:GHCi.DAP.Command
:dap-scopes ...
H>>= :dap-variables ...
```

Reply:
ok, I pushed the fix. Thanks for your testing!